sql/field.cc

```diff
@@ -7927,6 +7983,15 @@ Field *Field_varstring::new_key_field(MEM_ROOT *root, TABLE *new_table,
   return res;
 }
 
+static bool supports_such_enlargement(const Field_varstring *varchar,
+                                      Create_field *new_field)
+{
+  return varchar->field_length <= 127 || new_field->length <= 255 ||
+         varchar->field_length > 255 ||
+         (varchar->table->file->ha_table_flags() &
+          HA_EXTENDED_TYPES_CONVERSION);
+}
```

Comment:
I think that this function needs a `/** comment */` before it. Can you also explain in the comment why the comparisons with 127 and 255 are needed?